cartero: 0.2.2 -> 0.2.4

0xErwin1 opened the pull request:
Update Cartero to version 0.2.4.
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

0xErwin1 committed:
cartero: 0.2.2 -> 0.2.4

amerinor01 reviewed (commented):
Hi @0xErwin1,

First of all, thank you for your contribution! I’ve been reviewing the code, and I have a few questions and observations.

To the best of my knowledge, some of the new dependencies may not be necessary. For example, I believe graphene is already included as a Rust dependency and doesn’t need to be added separately.

Additionally, certain dependencies like gdk-pixbuf appear to be required only when building Cartero for macOS. Since we currently only support Linux platforms, these may not be needed at this time. (We can try to implement into darwin)

I also think there’s some duplication with OpenSSL. From what I can tell, the difference between the openssl and openssl_3 derivations is mainly the version. We should probably stick to just one—I think either would work, but unless there’s a compelling reason otherwise, I’d suggest using openssl as it appears to be more up to date.

Unfortunately, I won’t be able to test the GUI in the coming days, so I can only validate things through builds for now. If anything I’ve mentioned doesn’t work as expected, please let me know!

0xErwin1 commented:
@amerinor01  Thanks for the review.

I applied the suggested changes:
- Removed unnecessary `graphene` and `gdk-pixbuf` since they aren’t required on Linux.
- Unifed `openssl`/`openssl_3` to just `openssl`.
- Cleaned up duplicate inputs.

Initially, I based the update on the cartero's flake, so I added more dependencies than necessary.
Regarding macOS support. Unfortunately I can’t test or maintain a Darwin build since I don’t have a macOS environment. I’ll keep my focus on Linux (x86_64 and aarch64).

amerinor01 reviewed (changes requested):
Hi I'm back! :smile: 

Thanks once again for the contribution. I think there are still some duplicated dependencies.
Let me know if you need any help.

Here is the diff where I removed duplicates.
```diff
--- a/pkgs/by-name/ca/cartero/package.nix
+++ b/pkgs/by-name/ca/cartero/package.nix
@@ -49,14 +49,7 @@ stdenv.mkDerivation (finalAttrs: {
     wrapGAppsHook4
     desktop-file-utils
     libxml2
-    meson
-    ninja
     python313
-    cargo
-    rustc
-    pkg-config
-    blueprint-compiler
-    desktop-file-utils
     gtk4
     shared-mime-info
     glib
```

amerinor01 reviewed (approved):
LGTM

Aleksanaa commented on pkgs/by-name/ca/cartero/package.nix:
```diff
@@ -42,15 +48,26 @@ stdenv.mkDerivation (finalAttrs: {
   ];
 
+  postPatch = ''
+    patchShebangs build-aux/gen-version.py
```
```suggestion
    patchShebangs --build build-aux/gen-version.py
```

Aleksanaa reviewed (commented):
Please squash into one commit `cartero: 0.2.2 -> 0.2.4`
